lambdas/fetch-questions/src/fetch-questions-handler.ts

```diff
@@ -1,19 +1,36 @@
 import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Logger } from "@aws-lambda-powertools/logger";
+
+const logger = new Logger();
 
 export class FetchQuestionsHandler implements LambdaInterface {
   public async handler(event: any, _context: unknown): Promise<string> {
-    const response = await fetch(event.parameters.url, {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-        "User-Agent": event.parameters.userAgent,
-        Authorization: "Bearer " + event.bearerToken.value,
-      },
-      body: JSON.stringify({
-        nino: event.nino,
-      }),
-    });
-    return await response.json();
+    try {
+      const response = await fetch(event.parameters.url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "User-Agent": event.parameters.userAgent,
+          Authorization: "Bearer " + event.bearerToken.value,
+        },
+        body: JSON.stringify({
+          nino: event.nino,
+        }),
+      });
+      let json;
+      try {
+        json = await response.json();
+      } catch (error: any) {
+        return await response.text();
+      }
+      if (json.questions.length <= 0) {
+        throw new Error("No questions returned");
+      }
+      return json;
+    } catch (error: any) {
+      logger.error("Error in MatchingHandler: " + error.message);
+      throw error;
+    }
```

Comment:
Why do we have 2 catch blocks, should be doable with one catch block. If you assume a successful result response first you can throw No question returned if there are no questions. if something was wrong with the request it would be caught in the single catch block

Reply:
I noticed this originally in NINO Check, when HMRC give error messages, sometimes it's not a JSON format. If we throw the error like we did before, we would lose this information which is useful for debugging later. 

So by getting the response as a text, we can still handle do something with the error. `response.text()` can also throw which is then caught by the outer catch-block.

Reply:
I have updated it further, let me know what you think

Reply:
`json.questions.length` to `!json.questions || json.questions.length == 0` maybe

Reply:
I guess you can add in tests? I don't see any then this logic would be apparent

Reply:
Yeah we should do that when tests get added